Failed to fetch conversations.replies, when receiving a direct message

Hey there, thanks for the awesome library!

I'm actually using the `slack-cloudflare-workers` library, v1.3.1. But I believe it's consuming this underlying library, thus I'm creating the issue here. Do kindly inform me if there is a more optimal location for this 🙏 

I have a `SlackApp` that's listening to `event('message'`, in particular to receive Direct Messages for my Slack App.

When receiving this Direct Message, I see the following message logged before my handler can execute:

```
Failed to fetch conversations.replies API result: SlackAPIError: Failed to call conversations.replies due to invalid_arguments: {"ok":false,"error":"invalid_arguments","response_metadata":{"messages":["[ERROR] missing required field: ts"]},"headers":{}}
```

I assume the issue could be on this line? https://github.com/slack-edge/slack-edge/blob/233fdc3fdbfbb55a68c61df5ddfa470cb5b11148/src/assistant/thread-context-store.ts#L39
Sadly I lack context on this library to investigate further and send a PR myself 🙏 

Looking at the API method that I believe this to be calling: https://api.slack.com/methods/conversations.replies
I can confirm all of the related OAuth scopes are being requested and confirmed by my app. Thus, I do not consider this to be a permission issue (otherwise I'd also expect an authorization-related error from the API as well).

Thanks again! Do let me know if I can provide any additional information to assist with this 🙇

--- a/src/app.ts
+++ b/src/app.ts
@@ -232,6 +232,8 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
   #viewSubmissions: ((body: SlackRequestBody) => SlackViewHandler<E, ViewSubmission> | null)[] = [];
   #viewClosed: ((body: SlackRequestBody) => SlackViewHandler<E, ViewClosed> | null)[] = [];
 
+  #assistantEnabled: boolean;
+
   // --------------------------
 
   constructor(options: SlackAppOptions<E>) {
@@ -269,6 +271,7 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
     this.authorizeErrorHandler = options.authorizeErrorHandler ?? buildDefaultAuthorizeErrorHanlder();
     this.routes = { events: options.routes?.events };
     this.assistantThreadContextStore = options.assistantThreadContextStore;
+    this.#assistantEnabled = options.assistantThreadContextStore !== undefined;
   }
 
   /**
@@ -389,11 +392,12 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
     lazy: EventLazyHandler<Type, E>,
     handleSelfBotMessageEvents: boolean = false,
   ): SlackApp<E> {
+    this.#assistantEnabled = true;
     this.#events.push((body) => {
       if (body.type !== PayloadType.EventsAPI || !body.event) {
         return null;
       }
-      if (body.event.type === event && isAssitantThreadEvent(body)) {
+      if (body.event.type === event && this.#assistantEnabled && isAssitantThreadEvent(body)) {
         if (event === "message" && "bot_profile" in body.event) {
           if (handleSelfBotMessageEvents) {
             // this app's bot message events
@@ -874,7 +878,7 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
         const context = authorizedContext as SlackAppContextWithChannelId;
         const primaryToken = context.functionBotAccessToken || context.botToken;
         const client = new SlackAPIClient(primaryToken);
-        if (authorizedContext.isAssistantThreadEvent) {
+        if (this.#assistantEnabled && authorizedContext.isAssistantThreadEvent) {
           const assistantContext = authorizedContext as SlackAppContextWithAssistantUtilities;
           const { channelId: channel_id, threadTs: thread_ts } = assistantContext;
           // setStatus

--- a/src_deno/app.ts
+++ b/src_deno/app.ts
@@ -286,6 +286,8 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
   #viewClosed:
     ((body: SlackRequestBody) => SlackViewHandler<E, ViewClosed> | null)[] = [];
 
+  #assistantEnabled: boolean;
+
   // --------------------------
 
   constructor(options: SlackAppOptions<E>) {
@@ -332,6 +334,7 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
       buildDefaultAuthorizeErrorHanlder();
     this.routes = { events: options.routes?.events };
     this.assistantThreadContextStore = options.assistantThreadContextStore;
+    this.#assistantEnabled = options.assistantThreadContextStore !== undefined;
   }
 
   /**
@@ -469,11 +472,15 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
     lazy: EventLazyHandler<Type, E>,
     handleSelfBotMessageEvents: boolean = false,
   ): SlackApp<E> {
+    this.#assistantEnabled = true;
     this.#events.push((body) => {
       if (body.type !== PayloadType.EventsAPI || !body.event) {
         return null;
       }
-      if (body.event.type === event && isAssitantThreadEvent(body)) {
+      if (
+        body.event.type === event && this.#assistantEnabled &&
+        isAssitantThreadEvent(body)
+      ) {
         if (event === "message" && "bot_profile" in body.event) {
           if (handleSelfBotMessageEvents) {
             // this app's bot message events
@@ -1042,7 +1049,9 @@ export class SlackApp<E extends SlackEdgeAppEnv | SlackSocketModeAppEnv> {
         const context = authorizedContext as SlackAppContextWithChannelId;
         const primaryToken = context.functionBotAccessToken || context.botToken;
         const client = new SlackAPIClient(primaryToken);
-        if (authorizedContext.isAssistantThreadEvent) {
+        if (
+          this.#assistantEnabled && authorizedContext.isAssistantThreadEvent
+        ) {
           const assistantContext =
             authorizedContext as SlackAppContextWithAssistantUtilities;
           const { channelId: channel_id, threadTs: thread_ts } =

